src/lab1/Main.java

```diff
@@ -0,0 +1,179 @@
+        System.arraycopy(movies, 0, newMovies, 0, movies.length);
+        for (int i = movies.length; i < newMovies.length; i++) {
+            sc.nextLine();
+            System.out.print("Enter movie name: ");
+            String movieName = sc.nextLine();
+            System.out.print("Enter movie rating: ");
+            double movieRating = sc.nextDouble();
+            Movie movie = new Movie(movieName, movieRating);
+            newMovies[i] = movie;
+        }
+        return newMovies;
+    }
+
+    static double findAvgRating(Movie[] movies) {
+        double avgRating = 0;
+        for (Movie movie : movies) {
+            avgRating += movie.getRating();
+        }
+        avgRating = avgRating / movies.length;
+        return avgRating;
+    }
+
+    static void moviesWithMaxAndMinRating(Movie[] movies) {
+        double maxRating = movies[0].getRating();
+        double minRating = movies[0].getRating();
+        for (Movie movie : movies) {
+            if (movie.getRating() > maxRating) {
+                maxRating = movie.getRating();
+            }
+            if (movie.getRating() < minRating) {
+                minRating = movie.getRating();
+            }
+        }
+        System.out.println("The movie with the maximum rating: " + maxRating);
+        System.out.println("The movie with the minimum rating: " + minRating);
+    }
+
+    static void findMovie(String movieName,Movie[] movies) {
+        for (Movie movie : movies) {
+            if (movie.getName().equals(movieName)) {
+                System.out.println("Rating of movie \"" + movieName + "\" is " + movie.getRating());
+                return;
+            }
+        }
+        System.out.println("Movie \"" + movieName + "\" not found");
+    }
+
+    static void updateMovieRating(String movieName,Movie[] movies,Scanner sc) {
+        for (Movie movie : movies) {
+            if (movie.getName().equals(movieName)) {
+                System.out.print("Enter the new rating for the movie \"" + movieName + "\": ");
+                movie.setRating(sc.nextDouble());
+                return;
+            }
+        }
+        System.out.println("Movie \"" + movieName + "\" not found");
+    }
+
+    static Movie[] deleteMovie(String movieName,Movie[] movies) {
+        boolean found = false;
+        for (Movie movie : movies) {
+            if (movie.getName().equals(movieName)) {
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            System.out.println("Movie \"" + movieName + "\" not found");
+            return movies;
+        } else {
+            System.out.println("Movie \"" + movieName + "\" deleted");
+        }
+        Movie[] newMovies = new Movie[movies.length - 1];
+        for(int i = 0, j = 0; i < movies.length; i++, j++) {
+            if(movies[i].getName().equals(movieName)) {
+                j--;
+                continue;
+            }
+            newMovies[j] = movies[i];
```

Comment:
instead you can use System.arraycopy(); which you already use in addMovies method, because it is @IntrinsicCandidate (if you don't know about it, pls read)